server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalTopHits.java

```diff
@@ -184,9 +184,15 @@ protected boolean mustReduceOnSingleInternalAgg() {
     public Object getProperty(List<String> path) {
         if (path.isEmpty()) {
             return this;
-        } else {
-            throw new IllegalArgumentException("path not supported for [" + getName() + "]: " + path);
         }
+        assert path.size() == 1 : "property paths for top_hits can only contain a single field";
+        assert searchHits.getHits().length == 1 : "property paths should only resolve against top_hits with size == 1.";
+        Map<String, Object> sourceAsMap = searchHits.getAt(0).getSourceAsMap();
```

Comment:
I wonder whether other properties of search hit should be accessible? Like the score or sort values. 
Maybe the source should be accessible if `_source` as key is used?  For example `_source.field_name` to access a specific field in the source.

Reply:
I added support for specifying `_source.field`, as well as `_sort` and `_score`. Ptal, is this what you had in mind?